### PR TITLE
Examples of using the API

### PR DESCRIPTION
**Pull request by tellison**
Initial set of API examples for documentation.

# Checklist

- [ ] You added tests to cover the change
- [ ] `mvn clean install` build and test completes
- [ ] You changed or added to the documentation

### CODE REVIEW

**Comment by tellison** [comment]
Not sure if this should be here, or on the website?

**Comment by tellison** [comment]
> I think it is useful here and can be linked to from a simpler document on the website.

The website does a nicer job or asciidoc rendering than GitHub, but that's not a good reason for moving it ;-)

**Review by karianna — APPROVED** [review]
LGTM

File: docs/cookbook.adoc [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+----
+
+Again, the `API_URL` can be adapted for different Java versions etc. as shown in <<example-two,Example Two>>.
+
+
+== Example Three: More complex queries
+
+The API can be supplemented to filter the details response by Java version, platform, and other conditions and can be used to answer detailed queries.
+
+For example, to find the date of the most recent build of Java 19 on Windows 32-bit. The following API call will find all builds of Java 19 or later for Windows 32-bit architecture, sorted with most recent first.
```
Comment (johnoliver) [inline docs/cookbook.adoc:118]: The query you have `[19,]` would give you the most recent version after 19, so would include 20,21... etc

Comment (johnoliver) [reply docs/cookbook.adoc:118]: Oh, I see you already have that, its just the statements of:

"For example, to find the date of the most recent build of Java 19"

and

"find all builds of Java 19 or later"

are inconsistent here

File: docs/cookbook.adoc [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+[source,html]
+https://api.adoptium.net/v3/assets/version/%5B19%2C%5D?os=windows&architecture=x32&sort_method=DATE&sort_order=DESC
```
Comment (johnoliver) [inline docs/cookbook.adoc:121]: One subtility here that may not be obvious to users, sorting by DATE will give you the most recently released and not most recent version, i.e if your version range was 17+, then a 17 released recently would be higher up than a 19 version released a while ago.

The wording may be ambiguous here as people may think they are getting the date of the most recent version

File: docs/cookbook.adoc (hunk 1 above)
Comment (tellison) [reply docs/cookbook.adoc]: Thanks John, quite right. I have fixed that inconsistency.

File: docs/cookbook.adoc (hunk 2 above)
Comment (tellison) [reply docs/cookbook.adoc]: ACK. Clarified that is the release date.

**Review by johnoliver — APPROVED** [review]

**Pull request merged**